File: Todo.Web/Controllers/HomeController.vb
Imports Todo.Data.Services

Public Class HomeController
    Inherits System.Web.Mvc.Controller

    Private ReadOnly db As ITodoData
    Public Sub New()
        Me.db = New InMemoryData()
    End Sub

    Public ReadOnly Property Db1 As ITodoData
        Get
            Return db
        End Get
    End Property

    Function Index() As ActionResult
        Return View()
    End Function

    Function About() As ActionResult
        ViewData("Message") = "Your application description page."

        Return View()
    End Function

    Function Contact() As ActionResult
        ViewData("Message") = "Your contact page."

        Return View()
    End Function
End Class

File: Todo_Web/Controllers/HomeController.cs
using System;
using System.Collections.Generic;
using System.Configuration;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using Todo.Data.Services;
using Todo_Web.Models;

namespace Todo_Web.Controllers
{
    public class HomeController : Controller
    {
        ITodoData db;

        public HomeController()
        {
            db = new InMemoryData();
        }

        public ActionResult Index()
        {
            var model = new GreetingViewModel();
            model.Message = ConfigurationManager.AppSettings["message"];
            return View(model);
        }

        public ActionResult About()
        {
            ViewBag.Message = "Your application description page.";

            return View();
        }

        public ActionResult Contact()
        {
            ViewBag.Message = "Your contact page.";

            return View();
        }
    }
}